### ISSUE TITLE
Documentation on how to install dependencies

### ISSUE THREAD

**Issue by santagada** [mesonbuild/meson#1137]
Mesos docs says that if you depend on dependencies they will be resolved automatically, but I tried copying the example on https://github.com/mesonbuild/meson/tree/master/test%20cases/frameworks/3%20gmock and running mesos it only says dependency not found:

```
Dependency GTest found: NO

Meson encountered an error in file meson.build, line 6, column 0:
Dependency "gtest" not found
```

Is there any docs on how to make this work?

**Comment by jpakkane** [comment]
That means that the dependency is not available on your system. You need to add it with your distro package manager.

**Comment by santagada** [comment]
On macOS homebrew specifically doesn't come with it because you should directly download it yourself. I  thought menson had support for macos. Conan and others download the soyrce during configuration.

**Comment by QuLogic** [comment]
Conan is a package manager; meson is a build tool.

**Comment by nirbheek** [comment]
Meson is a build system, like Autotools, CMake, Waf, etc. It does not handle automatic downloading of system dependencies. For that, you will need something like Macports, Homebrew, etc. I'm not sure what you're expecting Meson to do here.

**Comment by santagada** [comment]
Yeah I got it confused. I came from more modern languages that don't have a differing concept between downloading dependencies and build system. CMake does have support for downloading dependencies by using external packages (and is was biicode, hunter and other extensions for it use) but is Meson doesn't have it its fine.

Maybe docs should be clear about it specially when for what I've seen in the source the "automatic" resolution for gtest is just looking for it on linux specific paths (on macOS if a package manager would install gtest it would have to put it on /usr/local/src not on /usr/src).

In sum there is no ExternalProject (https://cmake.org/cmake/help/v3.0/module/ExternalProject.html)  like in CMake?

**Comment by jpakkane** [comment]
If your system does not ship with gtest, then you can use [the wrap dependency system](https://github.com/mesonbuild/meson/wiki/Wrap%20dependency%20system%20manual) and [download it automatically from wrapdb](https://github.com/mesonbuild/meson/wiki/Using%20the%20WrapDB).

Note however that because gtest is "special" you still need to write some code to compile it with your project specific settings. We should probably write some docs on how to do that.

**Comment by santagada** [comment]
well as I got the impression that meson didn't download sources for compilation I just learned enough CMake import the code for now. But please do write the docs for it meson does seems much less complicated than CMake and I would like to switch in the future.

**Comment by bonzini** [comment]
Closing as there would be a book to write...

**Comment by santagada** [comment]
Thanks I can't remember what I was on about.
